src/pudl/output/eia923.py

```diff
@@ -204,19 +207,20 @@ def generation_fuel_all_eia923(gf: pd.DataFrame, gfn: pd.DataFrame) -> pd.DataFr
         raise ValueError(
             "Found inhomogeneous non-data cols while aggregating nuclear generation."
         )
-    agg_dict = dict(
-        **{col: "first" for col in non_data_cols},
-        **{col: "sum" for col in sum_cols},
-    )
-    gfn_agg = (
-        gfn_gb.agg(agg_dict)
-        .assign(fuel_mmbtu_per_unit=lambda x: x.fuel_consumed_mmbtu / x.fuel_consumed_units)
-        .reset_index()
-    )
-    # Nuclear plants don't report units of fuel consumed, so heat rate is infinite.
-    # However, some nuclear plants report using small amounts of DFO. Replace the
-    # infinite heat rates with NA:
-    gfn_agg.loc[np.isinf(gfn_agg.fuel_mmbtu_per_unit), "fuel_mmbtu_per_unit"] = np.nan
+    gfn_agg = pd.concat([
+        gfn_gb[non_data_cols].first(),
+        gfn_gb[sum_cols].sum(),
```

Comment:
can you add min_count=1 here?? if we ever fix the zero problem (i.e. null 0's  that are meant to represent nulls) this will cause zeros here.